lib/aws.go

```diff
@@ -0,0 +1,293 @@
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/common/types/traits"
+)
+
+// AWS returns a cel.EnvOption to configure extended functions for
+// AWS request signing. In all cases the functions are member overloads
+// on a <map<string,dyn>> which is expected to correspond to an HTTP
+// request, the returned value is the request with the necessary AWS v4
+// signing details added.
+//
+// # Sign AWS from env
+//
+// Returns a signed request based on credential details in the relevant
+// AWS environment variables; AWS_ACCESS_KEY, AWS_ACCESS_KEY_ID,
+// AWS_SECRET_ACCESS_KEY and AWS_SECRET_KEY.
+//
+//	<map<string,dyn>>.sign_aws_from_env(<string>, <string>, <timestamp>, <bool>, <bool>, <bool>) -> <map<string,dyn>>
+//
+// Examples:
+//
+//	req.sign_aws_from_env("service", "region", now(), false, false, false)   // return {"Body": "…
+//
+// The parameters to sign_aws_from_env correspond to the [v4_creds.Signer.Sign]
+// service, region and signTime parameters, while the three boolean parameters
+// correspond to the [v4_creds.Signer] fields; DisableHeaderHoisting,
+// DisableURIPathEscaping and UnsignedPayload respectively.
+//
+// This corresponds to using [v4_creds.Signer.Sign] using the result of a call
+// to [credentials.NewEnvCredentials].
+//
+// # Sign AWS from shared credentials
+//
+// Returns a signed request based on credential details in the relevant
+// credentials file, which may be pointed to by the AWS_SHARED_CREDENTIALS_FILE
+// and AWS_PROFILE environment variables.
+//
+//	<map<string,dyn>>.sign_aws_from_shared(<string>, <string>, <string>, <string>, <timestamp>, <bool>, <bool>, <bool>) -> <map<string,dyn>>
+//
+// Examples:
+//
+//	req.sign_aws_from_shared("filepath", "profile", "service", "region", now(), false, false, false)   // return {"Body": "…
+//
+// or with AWS_SHARED_CREDENTIALS_FILE set to "filepath" and AWS_PROFILE set to
+// "profile":
+//
+//	req.sign_aws_from_shared("", "", "service", "region", now(), false, false, false)   // return {"Body": "…
+//
+// The last six parameters are as described for sign_aws_from_env.
+//
+// This corresponds to using [v4_creds.Signer.Sign] using the result of a call
+// to [credentials.NewSharedCredentials].
+//
+// # Sign AWS from shared credentials
+//
+// Returns a signed request based on credential details provided statically.
+//
+//	<map<string,dyn>>.sign_aws_from_static(<string>, <string>, <string>, <string>, <string>, <timestamp>, <bool>, <bool>, <bool>) -> <map<string,dyn>>
+//	<map<string,dyn>>.sign_aws_from_static(<string>, <string>, <string>, <string>, <string>, <string>, <timestamp>, <bool>, <bool>, <bool>) -> <map<string,dyn>>
+//
+// Examples:
+//
+//	req.sign_aws_from_static("id", "secret", "token", "service", "region", now(), false, false, false)   // return {"Body": "…
+//	req.sign_aws_from_static("id", "secret", "token", "provider", "service", "region", now(), false, false, false)   // return {"Body": "…
```

Comment:
```suggestion
//	req.sign_aws_from_static("access_key_id", "secret_acces_key", "session_token", "provider", "service", "region", now(), false, false, false)   // return {"Body": "…
```
Maybe easy to comprehend 🤔. I will leave to you.

Reply:
I think I'm OK with what is there, and the signature is already too long, so I'd rather keep it as is.